File: ml_phylogeny_inference/src/get_rf_distance.r
# m_matschiner Sat May 5 17:37:00 CEST 2018

# Load the ape and phangorn libraries.
library("ape")
library("phangorn")

# Get the command-line arguments.
args <- commandArgs(trailingOnly = TRUE)
tree1_file_name <- args[1]
tree2_file_name <- args[2]

# Read the two trees.
tree1 <- read.tree(tree1_file_name)
tree2 <- read.tree(tree2_file_name)

# Calculate the Robinson-Foulds distance.
cat("\nRobinson-Foulds distance:", RF.dist(tree1, tree2), "\n\n")

File: ml_phylogeny_inference/src/get_rf_distance.py
# m_matschiner Mon May 28 18:53:32 CEST 2018

# Import libraries.
import sys
from ete3 import Tree

# Get the command line arguments.
tree1_file = open(sys.argv[1])
tree2_file = open(sys.argv[2])

t1_str = tree1_file.read()
t2_str = tree2_file.read()

t1 = Tree(t1_str)
t2 = Tree(t2_str)
rf = t1.robinson_foulds(t2, unrooted_trees=True)[0]
print rf
